api/jsonrpc/server.go

```diff
@@ -175,58 +175,66 @@ func (j *JSONRPCServer) Execute(
 	defer span.End()
 
 	actionRegistry := j.vm.ActionRegistry()
-	action, err := (*actionRegistry).Unmarshal(codec.NewReader(args.Action, len(args.Action)))
-	if err != nil {
-		return fmt.Errorf("failed to unmashal action: %w", err)
+	actions := make([]chain.Action, 0)
```

Comment:
same comment on pre-allocating where we already know the length of `args.Actions`